arch/arm/core/swap.S

```diff
@@ -324,6 +327,19 @@ _oops:
     blx _do_kernel_oops
     pop {pc}
 
+_escalate_priv:
+    /* check to see if we are already privileged */
+    mrs r2, CONTROL
+    tst r2, #0
+    bne _oops
+
+    /* set mode to privileged */
+    bic r2, #1
+    msr CONTROL, r2
+
+    /* fix PSP to point to privileged portion of stack */
+    bx lr
```

Comment:
I'm a little confused about how this works..

It seems that upon "svc #3" we land here, and if we are unprivileged, we set mode to privileged, switch to the kernel stack, and return wherever execution was when the "svc #3" call was made.

what's to prevent malicious user code from entering privileged mode anytime they please and then running whatever code they want as a supervisor?

Reply:
We can't do that unfortunately because we cannot handle nested SVC calls.  This would mean that we can't context switch without it faulting.